node-setup/GuideForExcercise1.md

```diff
@@ -0,0 +1,336 @@
+
+For Debian/Ubuntu use the following instead:
+   
+        
+    sudo apt-get update -y
+    sudo apt-get -y install build-essential pkg-config libffi-dev libgmp-dev libssl-dev libtinfo-dev libsystemd-dev zlib1g-dev make g++ tmux git jq wget libncursesw5 -y
+   
+If you are using a different flavor of Linux, you will need to use the package manager suitable for your platform instead of `yum` or `apt-get`, and the names of the packages you need to install might differ.
+
+Download, unpack, install and update Cabal:
+
+    wget https://downloads.haskell.org/~cabal/cabal-install-3.2.0.0/cabal-install-3.2.0.0-x86_64-unknown-linux.tar.xz
+    tar -xf cabal-install-3.2.0.0-x86_64-unknown-linux.tar.xz
+    rm cabal-install-3.2.0.0-x86_64-unknown-linux.tar.xz cabal.sig
+    mkdir -p ~/.local/bin
+    mv cabal ~/.local/bin/
+    cabal update
+
+This will work on a fresh [AWS instance](AWS.md) and assumes that folder `~/.local/bin` is in your `PATH`.
+On other systems, you must either move the executable to a folder that is in your `PATH` or modify your `PATH` by adding the line
+
+    export PATH="~/.local/bin:$PATH"
+
+to your `.bashrc`-file.
+
+Above instructions install Cabal version `3.2.0.0`. You can check the version by typing
+
+   cabal --version
+
+Finally, we download and install GHC:
+
+    wget https://downloads.haskell.org/~ghc/8.6.5/ghc-8.6.5-x86_64-deb9-linux.tar.xz
+    tar -xf ghc-8.6.5-x86_64-deb9-linux.tar.xz
+    rm ghc-8.6.5-x86_64-deb9-linux.tar.xz
+    cd ghc-8.6.5
+    ./configure
+    sudo make install
+    cd ..
+
+## Download the source code for cardano-node
+
+To download the source code, we use git:
+    
+    git clone https://github.com/input-output-hk/cardano-node.git
+    
+
+This should create a folder ``cardano-node``, then download the latest source code from git into it.
+After the download has finished, we can check its content by
+  
+    ls cardano-node
+
+__Note__ that the content of your `cardano-node`-folder can slightly differ from this!
+
+We change our working directory to the downloaded source code folder:
+
+    cd cardano-node
+
+For reproducible builds, we should check out a specific release, a specific "tag". 
+For the FF-testnet, we will use tag `pioneer`, which we can check out as follows:
+
+    git fetch --all --tags
+    git checkout tags/pioneer
+
+
+## Build and install the node
+
+Now we build and install the node with ``cabal``, 
+which will take a couple of minutes the first time you do a build. Later builds will be much faster, because everything that does not change will be cached.
+
+    cabal install cardano-node cardano-cli
+
+__Note__: At the time of writing, there is a bug in the latest version of the software that prevents ``cabal install`` from working correctly. As a workaround, you can use ``cabal build`` instead:
+
+    cabal build all
+    cp -p dist-newstyle/build/x86_64-linux/ghc-8.6.5/cardano-node-1.11.0/x/cardano-node/build/cardano-node/cardano-node ~/.local/bin/
```

Comment:
`cardano-node-1.11.0` could be `cardano-node-*` in order to work with any version
the same for `cardano-cli-1.11.0`